functionalDB/chapter.md

```diff
@@ -0,0 +1,1138 @@
+# Designing a database like an archaeologist
+
+Software development is often viewed as a rigorous process, where the inputs are requirements and the output is the working product. However, software developers are people with their own perspectives and biases which color the outcome of their work. 
+
+In this chapter, we will explore how a change in a common perspective affects the design and implementation of a well-studied type of software -- a database.
+
+## Introduction 
+
+Database systems are designed to store and query data. This is something that all information workers do; however, the systems themselves were designed by computer scientists. As a result, modern database systems are highly influenced by what a computer scientist’s definition of data is, and what can be done with it. 
+
+For example, most modern databases implement updates by overwriting old data in-place instead of appending the new data and keeping the old. This mechanism, nicknamed "place oriented programming" by Rich Hickey [REF], saves storage space but makes it impossible to retrieve the entire history of a particular record. This design decision reflects the computer scientist’s perspective that ‘history’ is less important than the price of its storage. 
+
+If you were to instead ask an archaeologist what should be done with the old data, the answer would be "hopefully, it’s just buried underneath".
+
+(Disclaimer: my understanding of a typical archaeologist is based on a few museum visits, several wikipedia articles, and watching the entire Indiana Jones series).
+
+### From archaeology to databases
+
+If we were to ask our friendly archaeologist to design a database, we might expect the requirements to reflect what would be found at an *excavation site:*
+
+* All data is found and cataloged at the site
+* Digging deeper will expose the state of things in times past 
+* Artifacts found at the same layer are from the same period
+* Each artifact will consist of state that it accumulated in different periods 
+
+For example, a wall may have Roman symbols drawn on it at on one layer, and in a lower layer there may be Greek symbols. Both these observations are recorded as part of the wall's state.

```

Comment:
I guess that this is archaeology as seen by a software guy (this is why I added that archaeology knowledge is based on Indiana Jones...). It doesn't add much, just the context that we're adding the notion of time and history to databasing